lib/group_domain_models/lib/src/models/course.dart

```diff
@@ -7,39 +7,39 @@
 // SPDX-License-Identifier: EUPL-1.2
 
 import 'package:common_domain_models/common_domain_models.dart';
-import 'package:meta/meta.dart';
 import 'package:design/design.dart';
+import 'package:meta/meta.dart';
 import 'package:sharezone_common/helper_functions.dart';
-import 'member_role.dart';
-import 'write_permissions.dart';
+
 import 'group_info.dart';
 import 'group_type.dart';
-import 'group_key.dart';
+import 'member_role.dart';
+import 'write_permissions.dart';
 
 class Course {
-  final String id, name, subject, abbreviation, sharecode, joinLink, meetingID;
-  final MemberRole myRole;
+  final String? id, name, subject, abbreviation, sharecode, joinLink, meetingID;
```

Comment:
Can't you make `id` non-nullable?

Reply:
```dart
  factory Course.create() {
    return Course._(
      id: null,
      name: "",
      subject: "",
```

Since e.g. in here you could just replace `null` with a placeholder value if you need to.